### PR TITLE
Please add the assignment operator to hip_bfloat16

### PR DESCRIPTION
**Pull request by neon60**
Please add the assignment operator and bool conversion to hip_bfloat16. It would be usefull in rocPRIM and hipCUB.

### CODE REVIEW

File: include/hip/hip_bfloat16.h [hunk 1]
```diff
@@ -97,6 +102,30 @@ struct hip_bfloat16
...
     }
 
+    #if !defined(__HIP_NO_BFLOAT16_CONVERSIONS__)
+    template<
+        typename T,
+        Enable_if_t<std::is_floating_point<T>{}>* = nullptr>
+    __host__ __device__
+    hip_bfloat16& operator=(T x)
+    {
+        data = static_cast<_Float16>(x);
```
Comment (vince-streamhpc) [inline include/hip/hip_bfloat16.h]: _Float16 will not cover whole possible range of bfloat16 correctly
```suggestion
        data = float_to_bfloat16(x);
```

File: include/hip/hip_bfloat16.h [hunk 2]
```diff
@@ -97,6 +102,30 @@ struct hip_bfloat16
...
+    #endif
+
+    // MANIPULATORS - DEVICE ONLY
+    #if !defined(__HIP_NO_BFLOAT16_CONVERSIONS__)
+    template<
+        typename T, Enable_if_t<std::is_integral<T>{}>* = nullptr>
+    __host__ __device__
+    hip_bfloat16& operator=(T x)
+    {
+        data = static_cast<_Float16>(x);
```
Comment (vince-streamhpc) [inline include/hip/hip_bfloat16.h]: _Float16 will not cover whole possible range of bfloat16 correctly
```suggestion
        data = float_to_bfloat16(static_cast<float>(x));
```

**Review by vince-streamhpc — CHANGES_REQUESTED** [review]

File: include/hip/hip_bfloat16.h [hunk 3]
```diff
@@ -97,6 +102,30 @@ struct hip_bfloat16
         return output;
     }
 
+    #if !defined(__HIP_NO_BFLOAT16_CONVERSIONS__)
+    template<
+        typename T,
+        Enable_if_t<std::is_floating_point<T>{}>* = nullptr>
+    __host__ __device__
+    hip_bfloat16& operator=(T x)
```
Comment (vince-streamhpc) [inline include/hip/hip_bfloat16.h]:
```suggestion
    hip_bfloat16& operator=(const T& x)
```

File: include/hip/hip_bfloat16.h [hunk 4]
```diff
@@ -97,6 +102,30 @@ struct hip_bfloat16
...
+        return *this;
+    }
+    #endif
+
+    // MANIPULATORS - DEVICE ONLY
+    #if !defined(__HIP_NO_BFLOAT16_CONVERSIONS__)
+    template<
+        typename T, Enable_if_t<std::is_integral<T>{}>* = nullptr>
+    __host__ __device__
+    hip_bfloat16& operator=(T x)
```
Comment (vince-streamhpc) [inline include/hip/hip_bfloat16.h]:
```suggestion
    hip_bfloat16& operator=(const T& x)
```

**Review by vince-streamhpc — CHANGES_REQUESTED** [review]

File: include/hip/hip_bfloat16.h [hunk 5]
```diff
@@ -97,6 +102,30 @@ struct hip_bfloat16
...
+        Enable_if_t<std::is_floating_point<T>{}>* = nullptr>
+    __host__ __device__
+    hip_bfloat16& operator=(const T& x)
+    {
+        data = float_to_bfloat16(x);
+        return *this;
+    }
+    #endif
+
+    // MANIPULATORS - DEVICE ONLY
```
Comment (vince-streamhpc) [inline include/hip/hip_bfloat16.h:117]: comment not applicable?

File: include/hip/hip_bfloat16.h [hunk 6]
```diff
@@ -97,6 +102,30 @@ struct hip_bfloat16
...
+    #if !defined(__HIP_NO_BFLOAT16_CONVERSIONS__)
```
Comment (vince-streamhpc) [inline include/hip/hip_bfloat16.h:118]:
```suggestion
   
```
merge `#if` statements

**Review by vince-streamhpc — CHANGES_REQUESTED** [review]

**Comment by neon60** [comment]
We will change the rocPRIM and we will handle the bfloat16 without the extra assigment operator.

**Pull request closed without merging**